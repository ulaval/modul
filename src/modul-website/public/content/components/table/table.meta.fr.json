{
    "name": "Tableau",
    "url": "tableau",
    "category": "tables",
    "type": "components",
    "beta": false,
    "visible": true,
    "preview": "components/table/table.preview.fr.md",
    "components": [
        "m-table",
        "m-table-header"
    ]
}
