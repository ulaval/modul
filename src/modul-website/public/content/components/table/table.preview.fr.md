<modul-preview>

```javascript
{
    data: {
        columns: [
                    { id: 'name', title: 'Name', dataProp: 'name', sortable: true},
                    { id: 'age', title: 'Age', dataProp: 'age' , sortable: true},
                    { id: 'username', title: 'Username', dataProp: 'username', sortable: true }
                ],
         rows: [
                    { id: '1', name: 'Jonathan', age: '25', username: 'jonathan.25' },
                    { id: '2', name: 'Carl', age: '30', username: 'carl.30' },
                    { id: '3', name: 'Jacob', age: '26', username: 'jacob.26' },
                    { id: '4', name: 'Vincent', age: '34', username: 'vincent.34' },
                    { id: '5', name: 'Manon', age: '28', username: 'manon.28' }
                ]

    },
    methods: {
        onSortApplied: function(columnTable) {
             this.$data.rows.sort((a, b) => {
                if (a[columnTable.dataProp] < b[columnTable.dataProp]) {
                    return -1 * columnTable.sortDirection;
                } else if (a[columnTable.dataProp] > b[columnTable.dataProp]) {
                    return 1 * columnTable.sortDirection;
                }
                return 0;
            });
        }
    }

}
```

```html
<m-table :columns="columns" :rows="rows" width="100%" @sort-applied="onSortApplied($event)"></m-table>
```

</modul-preview>
