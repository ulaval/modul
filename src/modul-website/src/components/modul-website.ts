import { ModulVue } from '@ulaval/modul-components/dist/utils/vue/vue';

export class ModulWebsite extends ModulVue {

}
