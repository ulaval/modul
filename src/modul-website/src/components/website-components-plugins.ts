import Vue, { PluginObject } from 'vue';
import { MWCard, MWCARD_NAME } from './card/card';
import { COLOR_NAME, MColor } from './color/color';
import { COMPONENT_API_NAME, MComponentApi } from './component-api/component-api';
import { DEMO_NAME, MDemo } from './demo/demo';
import { DO_NAME, MDo } from './do/do';
import { DONT_NAME, MDont } from './dont/dont';
import { MDynamicTemplate } from './dynamic-template/dynamic-template';
import { MWExpandablePanel } from './expendable-panel/expandable-panel';
import { GO_NAME, MGo } from './go/go';
import { MWHeader, MWHEADER_NAME } from './header/header';
import { HIGHLIGHT_NAME, MHighlight } from './highlight/highlight';
import { ICON_GALLERY_NAME, MIconGallery } from './icon-gallery/icon-gallery';
import { MWLogo, MWLOGO_NAME } from './logo/logo';
import { MARKDOWN_NAME, MMarkdown } from './markdown/markdown';
import { MWMegaMenu, MWMEGAMENU_NAME } from './mega-menu/mega-menu';
import { MWPanel, MWPANEL_NAME } from './panel/panel';
import { MPreview, PREVIEW_NAME } from './preview/preview';
import SliderPlugin from './slider/slider';

const WebsiteComponentsPlugin: PluginObject<any> = {
    install(v): void {
        Vue.component(COLOR_NAME, MColor);
        Vue.component(DEMO_NAME, MDemo);
        Vue.component(DO_NAME, MDo);
        Vue.component(DONT_NAME, MDont);
        Vue.component(GO_NAME, MGo);
        Vue.component(MARKDOWN_NAME, MMarkdown);
        Vue.component(PREVIEW_NAME, MPreview);
        Vue.component(MWHEADER_NAME, MWHeader);
        Vue.component(MWMEGAMENU_NAME, MWMegaMenu);
        Vue.component(MWLOGO_NAME, MWLogo);
        Vue.component(MWCARD_NAME, MWCard);
        Vue.component(MWPANEL_NAME, MWPanel);
        Vue.component(HIGHLIGHT_NAME, MHighlight);
        Vue.component(ICON_GALLERY_NAME, MIconGallery);
        Vue.component(COMPONENT_API_NAME, MComponentApi);
        Vue.component('mw-dynamic-template', MDynamicTemplate);
        Vue.component('mw-expandable-panel', MWExpandablePanel);
        Vue.component('mw-expandable-panel', MWExpandablePanel);
        Vue.use(SliderPlugin);
    }
};

export default WebsiteComponentsPlugin;
