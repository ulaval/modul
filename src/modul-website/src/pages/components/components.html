<div class="mw-components">
    <m-expandable-layout class="mw-components__body"
                         :open="showMenu"
                         panel-width="320px">
        <div class="mw-components__page">
            <m-icon-button class="mw-components__menu-button"
                           @click="toggleMenu()"
                           :icon-name="showMenu ? 'm-svg__close-clear' : 'm-standard-edit'"
                           skin="light"
                           icon-size="20px">
            </m-icon-button>

            <div class="mw-components__content">
                <mw-component-details :component="component"></mw-component-details>
            </div>
        </div>
        <template slot="panel">
            <div class="mw-components__sidemenu">

                <div class="mw-components__search">
                    <m-searchfield></m-searchfield>
                </div>
                <div class="mw-components__sidemenu-body">
                    <div class="mw-components__sidemenu-title">
                        <m-i18n k="website:components-ui"></m-i18n>
                    </div>
                    <m-accordion skin="plain"
                                 icon-position="right"
                                 icon-size="small"
                                 v-for="categoryId in componentCategoryIds"
                                 :key="categoryId"
                                 :open="urlMatchPath(categoryId)">
                        <div class="mw-components__sidemenu-sub-title"
                             slot="header">
                            <m-i18n :k="'categories:'+categoryId"></m-i18n>
                        </div>
                        <div>
                            <div class="mw-components__sidemenu-item"
                                 v-for="component in getCategoryComponents(categoryId)"
                                 :key="component.url">
                                <router-link :to="getRouterIndex(component.url)"
                                             @click.native="onSideMenuSelection()"><span>
                                        {{component.name}}
                                    </span>
                                    <span v-if="component.beta">
                                        <span class="modul-logo__beta mw-components__beta">
                                            <m-i18n k="website:beta"></m-i18n>
                                        </span>
                                    </span>
                                </router-link>
                            </div>
                        </div>
                    </m-accordion>
                </div>
            </div>
        </template>
    </m-expandable-layout>
</div>
