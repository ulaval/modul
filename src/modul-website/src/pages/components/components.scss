@import '../../styles/commons';

.mw-component-details {
    width: 100%;

    &__component-navigation {
        display: flex;
        margin-top: 54px;
        color: $m-color--interactive-light;
    }

    &__component-navigation-next,
    &__component-navigation-prev {
        line-height: 24px;
        position: relative;
        padding: 1rem;
        flex-grow: 1;
        border: 1px solid;
        text-align: center;
        text-decoration: none;
        font-weight: $m-font-weight--semi-bold;
    }

    &__component-navigation-next--icon {
        position: absolute;
        right: 16px;
    }

    &__component-navigation-prev--icon {
        position: absolute;
        left: 16px;
    }
}
