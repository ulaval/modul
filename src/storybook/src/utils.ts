export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
