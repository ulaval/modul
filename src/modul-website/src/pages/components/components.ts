
import { MWExpandablePanel } from '@/components/expendable-panel/expandable-panel';
import { ModulWebsite } from '@/components/modul-website';
import { ComponentMeta } from '@/content/components.meta.loader';
import _ from 'lodash';
import Vue from 'vue';
import Component from 'vue-class-component';
import { Prop, Watch } from 'vue-property-decorator';
import { MWComponent } from './component/component-details';
import WithRender from './components.html?style=./components.scss';

@WithRender
@Component({
    components: { 'mw-component-details': MWComponent }
})
export class MWComponentsPage extends ModulWebsite {

    @Prop()
    component: ComponentMeta;

    $refs: {
        expandablePanel: MWExpandablePanel;
        prevLink: Vue;
        nextLink: Vue;
    };

    fullPath: string;

    get componentCategoryIds(): string[] {
        return _.sortBy(this.$meta.categories, name => _.deburr(this.$i18n.translate(`categories:${name}`)));
    }

    get componentsNavigationOrder(): ComponentMeta[] {
        let components: ComponentMeta[] = [];
        this.componentCategoryIds.forEach(category => {
            this.$meta.componentState[category].forEach((componentMeta: ComponentMeta) => {
                components.push(componentMeta);
            });
        });
        return components;
    }

    get currentComponentNavigationIndex(): number {
        return _.findIndex(this.componentsNavigationOrder, (c: ComponentMeta) => {
            return this.component.name === c.name;
        });
    }

    get prevComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the first item return the last
        if (index - 1 < 0) {
            return this.componentsNavigationOrder[this.componentsNavigationOrder.length - 1];
        } else {
            return this.componentsNavigationOrder[index - 1];
        }
    }

    get nextComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the last item return the first
        if (index + 1 >= this.componentsNavigationOrder.length - 1) {
            return this.componentsNavigationOrder[0];
        } else {
            return this.componentsNavigationOrder[index + 1];
        }
    }

    getRouterIndex(tag: string): string {
        return this.$routerIndex.for(tag);
    }

    getCategoryComponents(category): ComponentMeta[] {
        return this.$meta.componentState[category];
    }

    urlMatchPath(categoryId: string) {
        let path: string = this.$i18n.translate(`categories:${categoryId}-route`);
        return this.fullPath.includes(`/${path}`);
    }

    onSideMenuSelection() {
        this.$refs.expandablePanel.onSideMenuSelection();
    }

    onPrevLinkClick() {
        (this.$refs.prevLink.$el as HTMLElement).blur();
    }

    onNextLinkClick() {
        (this.$refs.nextLink.$el as HTMLElement).blur();
    }

    @Watch('$route', { immediate: true })
    private routeChanged(): void {
        this.fullPath = this.$route.fullPath;
    }
}
