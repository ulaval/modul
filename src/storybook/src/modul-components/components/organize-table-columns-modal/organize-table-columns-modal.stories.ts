import { actions } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/vue';
import { ORGANIZE_TABLE_COLUMNS_MODAL_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MColumnSortDirection, MColumnTable } from '@ulaval/modul-components/dist/components/table/table';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

function defaultOnSortApplied(columnTable: MColumnTable): void {
    if (columnTable.sortDirection === MColumnSortDirection.None) {
        this.$data.rows.sort((a, b) => a.id < b.id ? -1 : a.id > b.id ? 1 : 0);
        return;
    }

    this.$data.rows.sort((a, b) => {
        if (a[columnTable.dataProp] < b[columnTable.dataProp]) {
            return -1 * columnTable.sortDirection!;
        } else if (a[columnTable.dataProp] > b[columnTable.dataProp]) {
            return 1 * columnTable.sortDirection!;
        }
        return 0;
    });
}

storiesOf(`${modulComponentsHierarchyRootSeparator}${ORGANIZE_TABLE_COLUMNS_MODAL_NAME}`, module)
    .add('Default', () => ({
        props: {
            columns: {
                default: [
                    { id: 'check', title: 'Action', dataProp: 'check', width: '16px' },
                    { id: 'name', title: 'Name', dataProp: 'name' },
                    { id: 'age', title: 'Age', dataProp: 'age' },
                    { id: 'username', title: 'Username', dataProp: 'username' },
                    { id: 'menu', title: 'Menu', dataProp: 'menu', width: '33px' }
                ]
            }

        },
        methods: actions(
            'reorganize',
            'reset'
        ),
        template: `<m-organize-table-columns-modal @reorganize="reorganize" @reset=reset :columns="columns"></m-organize-table-columns-modal>`
    }))
    .add('Fixed columns', () => ({
        props: {
            columns: {
                default: [
                    { id: 'check', title: 'Action', dataProp: 'check', width: '16px', fixed: true },
                    { id: 'name', title: 'Name', dataProp: 'name', fixed: true },
                    { id: 'age', title: 'Age', dataProp: 'age' },
                    { id: 'username', title: 'Username', dataProp: 'username' },
                    { id: 'menu', title: 'Menu', dataProp: 'menu', width: '33px' }
                ]
            }
        },
        methods: actions(
            'reorganize',
            'reset'
        ),
        template: `<m-organize-table-columns-modal @reorganize="reorganize" @reset=reset :columns="columns"></m-organize-table-columns-modal>`
    }))
    .add('Hidden columns', () => ({
        props: {
            columns: {
                default: [
                    { id: 'check', title: 'Action', dataProp: 'check', width: '16px' },
                    { id: 'name', title: 'Name', dataProp: 'name' },
                    { id: 'age', title: 'Age', dataProp: 'age', visible: false },
                    { id: 'username', title: 'Username', dataProp: 'username' },
                    { id: 'menu', title: 'Menu', dataProp: 'menu', width: '33px' }
                ]
            }
        },
        methods: actions(
            'reorganize',
            'reset'
        ),
        template: `<m-organize-table-columns-modal @reorganize="reorganize" @reset=reset :columns="columns"></m-organize-table-columns-modal>`
    }))
    .add('Ignore columns', () => ({
        props: {
            columns: {
                default: [
                    { id: 'check', title: 'Action', dataProp: 'check', width: '16px' },
                    { id: 'name', title: 'Name', dataProp: 'name' },
                    { id: 'age', title: 'Age', dataProp: 'age', ignored: true },
                    { id: 'username', title: 'Username', dataProp: 'username' },
                    { id: 'menu', title: 'Menu', dataProp: 'menu', width: '33px' }
                ]
            }
        },
        methods: actions(
            'reorganize',
            'reset'
        ),
        template: `<m-organize-table-columns-modal @reorganize="reorganize" @reset=reset :columns="columns"></m-organize-table-columns-modal>`
    }));
