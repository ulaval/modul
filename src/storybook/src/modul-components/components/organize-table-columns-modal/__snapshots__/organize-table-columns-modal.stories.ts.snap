// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-organize-table-columns-modal Custom trigger button 1`] = `
<div
  class="organize-table-columns-modal"
>
  <div
    class="m-modal"
  >
    <button
      class="m-button m--is-skin-secondary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        test 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots modul-components|m-organize-table-columns-modal Default 1`] = `
<div
  class="organize-table-columns-modal"
>
  <div
    class="m-modal"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Réorganiser les colonnes 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots modul-components|m-organize-table-columns-modal Fixed columns 1`] = `
<div
  class="organize-table-columns-modal"
>
  <div
    class="m-modal"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Réorganiser les colonnes 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots modul-components|m-organize-table-columns-modal Hidden columns 1`] = `
<div
  class="organize-table-columns-modal"
>
  <div
    class="m-modal"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Réorganiser les colonnes 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots modul-components|m-organize-table-columns-modal Ignore columns 1`] = `
<div
  class="organize-table-columns-modal"
>
  <div
    class="m-modal"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Réorganiser les colonnes 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;
